src/sizes/retained_size_action.h

```diff
@@ -85,23 +86,23 @@ class RetainedSizeAction : public MemoryAgentTimedAction<RESULT_TYPE, jobjectArr
     jvmtiError tagHeap() {
         jvmtiError err = this->FollowReferences(0, nullptr, nullptr, getTagsWithNewInfo, nullptr, "find objects with new info");
         if (err != JVMTI_ERROR_NONE) return err;
-        if (this->shouldStopExecution()) return MEMORY_AGENT_TIMEOUT_ERROR;
+        if (this->shouldStopAction()) return MEMORY_AGENT_INTERRUPTED_ERROR;
```

Comment:
Macroses are something to avoid, but they seem pretty suitable for us now. Something like:
```
#define checkCanceled error = this->shouldStopAction(); if (error != CONTINUE) { return error; }
```

Reply:
if we want to refactor these checks with macroses, we have to provide 3 different ones:
```
// For inheritors of MemoryAgentTimedAction
#define checkCanceled error = this->shouldStopAction(); if (error != CONTINUE) { return error; } 

// For static functions that are used by many actions and want to check cancellation
#define checkCanceled(timeoutInMillis, cancellationFileName) error = shouldStopAction(timeoutInMillis, cancellationFileName); if (error != CONTINUE) { return error; } 

// For JVMTI error checks 
#define checkError(err) if (error != JVMTI_ERROR_NONE) return error
```
Won't it make the code messy?